### PR TITLE
[fix][train] Shut down DataLoader workers on exit

### PR DESCRIPTION
**Pull request by bvolpato**
## Summary

`StatefulDataLoader` owns worker processes through its iterator. RL and SFT entrypoints only cleaned up dataloaders on successful runs, so training exceptions could leave train or evaluation workers alive.

## Changes

- Add a guarded, idempotent shutdown helper for torchdata 0.11.
- Shut down RL train and evaluation dataloaders on success and failure.
- Move SFT shutdown into the entrypoint's `finally` block and close every train and evaluation dataloader.
- Log cleanup failures without masking the original training exception.

Normal Python unwinding is covered. Abrupt termination such as `SIGKILL` remains outside scope.

## Testing

- Focused lifecycle tests (`123 passed`)
- Full prior SkyRL Train CPU suite (`1467 passed, 5 skipped, 5 deselected`)
- Tests start real train and evaluation workers, call shutdown twice, and verify every worker exits
- `pre-commit run --all-files`
- `git diff --check upstream/main`

### CODE REVIEW

File: skyrl/train/sft_trainer.py [hunk 1]
```diff
@@ -2245,13 +2246,22 @@ def save_hf_model(self):
...
         the Ray lifecycle.
         """
+        dataloaders = [("train", self.train_dataloader)]
+        if self.eval_dataloaders is not None:
+            dataloaders.extend((f"eval/{name}", dataloader) for name, dataloader in self.eval_dataloaders)
+        for name, dataloader in dataloaders:
+            try:
+                shutdown_dataloader(dataloader)
+            except Exception:
+                logger.exception(f"Failed to shut down {name} DataLoader workers")
```
[the remark this answers is not in the document]
Comment (bvolpato) [reply skyrl/train/sft_trainer.py]: Updated SFT shutdown to handle train and named evaluation dataloaders in separate guarded blocks.
